curvefs/src/mds/fs_manager.cpp

```diff
@@ -171,6 +176,65 @@ void FsManager::BackEndFunc() {
     }
 }
 
+void MountPoint2Str(const Mountpoint& in, std::string *out) {
+    *out = in.hostname() + ":" + std::to_string(in.port()) + ":" + in.path();
+}
+
+bool Str2MountPoint(const std::string& in, Mountpoint* out) {
+    std::vector<std::string> vec;
+    curve::common::SplitString(in, ":", &vec);
+    if (vec.size() != 3) {
+        LOG(ERROR) << "split string to mountpoint failed, str = " << in;
+        return false;
+    }
+    out->set_hostname(vec[0]);
+    uint32_t port;
+    if (!curve::common::StringToUl(vec[1], &port)) {
+        LOG(ERROR) << "StringToUl failed, str = " << vec[1];
+        return false;
+    }
+    out->set_port(port);
+    out->set_path(vec[2]);
+    return true;
+}
+
+void FsManager::CheckMountPoint() {
+    WriteLockGuard wlock(recorderMutex_);
+    uint64_t now = ::curve::common::TimeUtility::GetTimeofDaySec();
+    auto iter = mpTimeRecorder_.begin();
+    while (iter != mpTimeRecorder_.end()) {
```

Comment:
When the map is very large, since you have taken the mutex lock here, all records cannot be updated, which will cause all leases to time out.

Reply:
fixed